tuning/sft_trainer.py

```diff
@@ -78,20 +78,27 @@ def train(
         model_args.model_name_or_path,
         cache_dir=train_args.cache_dir,
         model_max_length=train_args.model_max_length,
-        padding_side="right",
         use_fast = True
     )
+
+    # TODO: understand if we need to hardcode these here or just use defaults in model
     if isinstance(tokenizer, LlamaTokenizer) or isinstance(tokenizer, LlamaTokenizerFast):
         tokenizer.add_special_tokens({
             "bos_token": "<s>",
             "eos_token": "</s>",
             "unk_token": "<unk>",
             "pad_token": "<pad>",
         })
-    elif isinstance(tokenizer, GPTNeoXTokenizerFast):
+    if isinstance(tokenizer, GPTNeoXTokenizerFast) or isinstance(tokenizer, GPT2Tokenizer):
```

Comment:
yup will change it. also just for clarification I added gPT2 here because granite 20B models are using GPT2. But I think in cleanup we should get rid of thsi block entirely and just do if pad token unset , set it to default as done below regardless of type of tokenizer

Reply:
done